apps/site-landing/src/containers/BenefitsSimulator/Graph/index.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useState } from 'react';
+import { Header, Loader, Progress, Segment } from 'semantic-ui-react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import { AreaGraph } from '../../AreaGraph';
+import { calcAllReturns, calculateAvgAndArea, CoinReturns, MarketData, SimulationParameters } from '../../ReturnProfile/data';
+import styles from './styles.module.less';
+
+const translations = defineMessages({
+  title : {
+      defaultMessage: 'Your possible benefits',
+      description: 'site.compare.graph.title: Graph title for the How much will you make? graph page'},
+  description : {
+    defaultMessage: 'Your chequing account will be with you longer than your retirement savings will. This is what that could look like.',
+    description: 'site.compare.graph.description: Graph description for the How much will you make? graph page'}
+  });
+
+type Props = {
+  params: SimulationParameters,
+  fxData?: MarketData[],
+  snpData?: MarketData[],
+}
+
+// TODO: This component does a lot of computation, and should be memoized
+export const BenefitsGraph = ({params, snpData}: Props) => {
+
+  //const [allReturns, setAllReturns] = useState<undefined|SimulationState[][]>();
+  const [averages, setAverages] = useState<undefined|CoinReturns[]>();
+  const [progress, setProgress] = useState<number|undefined>(0);
+
+  useEffect(() => {
+    if (!snpData) return;
+    let isCancelled = false;
+    const cancellableProgress = (value: number) => {
+      setProgress(value);
+      return !isCancelled;
+    }
+    const allReturns = calcAllReturns(snpData, params, cancellableProgress);
+    if (allReturns) {
```

Comment:
possible UX-improvement.  Instead of blocking for minutes at a time, we could use generators to start displaying the results as quickly as possible